Bestiary work

Wembler23 opened the pull request:
Fixed a lot of logic for the specific entries, added back and forth buttons on the entries and the equipment views, changed everything to adventure.

Wembler23 committed:
logic fixes! hooray for logic

changed logic around damage mostly, so much stuff was wrong

Combustible committed:
Add Peer information to enable going forward/back through entries

Signed-off-by: Byron Marohn <Combustible@live.com>

Wembler23 committed:
Logic changes like you would not believe

Fixed way to many problems with my logic, added back and forth buttons, fixed more problems with my logic, transferred everything over to adventure.

Wembler23 committed:
Style changes

Wembler23 committed:
Fixed another isBlank()

Combustible commented on plugin/src/main/java/com/playmonumenta/libraryofsouls/bestiary/BestiarySoulInventory.java:
```diff
@@ -174,28 +234,51 @@ private static String formatWell(String in) {
 
 	private final SoulEntry mSoul;
 	private final BestiaryArea mParent;
+	/* TODO: Use these to provide previous / next buttons */
```
Comment to remove

Combustible commented on plugin/src/main/java/com/playmonumenta/libraryofsouls/bestiary/BestiarySoulInventory.java:
```diff
@@ -174,28 +234,51 @@ private static String formatWell(String in) {
+		DamageType type = null;
+
+		//Stuff to throw errors before everything
+		if (mDefaultHealth.get(entityNBT.getEntityType()) != null && health == 0.0) {
+			health += mDefaultHealth.get(entityNBT.getEntityType());
```
This pattern in general should be avoided - use .get() once, and store the value, then use it in both places.

Combustible commented on plugin/src/main/java/com/playmonumenta/libraryofsouls/bestiary/BestiarySoulInventory.java:
```diff
@@ -174,28 +234,51 @@ private static String formatWell(String in) {
+			LibraryOfSouls.getInstance().getLogger().log(Level.INFO, "This mob type is not contained in the health map: " + entityNBT.getEntityType());
+		}
+
+		if (mDefaultDamage.get(entityNBT.getEntityType()) != null && damage == 0.0) {
+			damage += mDefaultDamage.get(entityNBT.getEntityType());
```
Same as above - prefer only using one lookup

Combustible reviewed (approved):
Looks awesome, couple small requests

Wembler23 committed:
Merge remote-tracking branch 'origin/master' into wembler-bestiary-work

Wembler23 committed:
Changed logic and the color of basic mobs

Changed the color of basic mobs to black for ease of reading, as well as changed the text on the item to not be italicized and formatted the entity types properly.  Also cut down on how much I would regrab something like an item or entity's type / hash map objects.

Wembler23 committed:
Style changes

Combustible commented on plugin/src/main/java/com/playmonumenta/libraryofsouls/bestiary/BestiarySoulInventory.java:
```diff
@@ -346,44 +363,54 @@ public BestiarySoulInventory(Player player, SoulEntry soul, BestiaryArea parent,
+			} else {
+				String get = nbtVar.get();
+				if (get != null) {
+					explodePower = Double.valueOf(entityNBT.getVariable("ExplosionRadius").get());
```
probably meant `Double.valueOf(get)`

Wembler23 committed:
Fixed a line I missed in upgrading the get stuff
